src/sentry/eventstore/base.py

```diff
@@ -7,64 +7,72 @@
 
 
 class Columns(Enum):
-    # TODO add all the other columns.
-    EVENT_ID = "event_id"
-    GROUP_ID = "group_id"
-    ISSUE = "issue"
-    PROJECT_ID = "project_id"
-    TIMESTAMP = "timestamp"
-    CULPRIT = "culprit"
-    LOCATION = "location"
-    MESSAGE = "message"
-    PLATFORM = "platform"
-    TITLE = "title"
-    TYPE = "type"
-    TAGS_KEY = "tags.key"
-    TAGS_VALUE = "tags.value"
-    EMAIL = "email"
-    IP_ADDRESS = "ip_address"
-    USER_ID = "user_id"
-    USERNAME = "username"
-    TRANSACTION = "transaction"
-    USER_ID = "user_id"
-    USER_EMAIL = "email"
-    USER_USERNAME = "username"
-    USER_IP = "ip_address"
-    SDK_NAME = "sdk_name"
-    SDK_VERSION = "sdk_version"
-    HTTP_METHOD = "http_method"
-    HTTP_REFERER = "http_referer"
-    HTTP_URL = "http_url"
-    OS_BUILD = "os_build"
-    OS_KERNEL_VERSION = "os_kernel_version"
-    DEVICE_NAME = "device_name"
-    DEVICE_BRAND = "device_brand"
-    DEVICE_LOCALE = "device_locale"
-    DEVICE_UUID = "device_uuid"
-    DEVICE_ARCH = "device_arch"
-    DEVICE_BATTERY_LEVEL = "device_battery_level"
-    DEVICE_ORIENTATION = "device_orientation"
-    DEVICE_SIMULATOR = "device_simulator"
-    DEVICE_ONLINE = "device_online"
-    DEVICE_CHARGING = "device_charging"
-    GEO_COUNTRY_CODE = "geo_country_code"
-    GEO_REGION = "geo_region"
-    GEO_CITY = "geo_city"
-    ERROR_TYPE = "exception_stacks.type"
-    ERROR_VALUE = "exception_stacks.value"
-    ERROR_MECHANISM = "exception_stacks.mechanism_type"
-    ERROR_HANDLED = "exception_stacks.mechanism_handled"
-    STACK_ABS_PATH = "exception_frames.abs_path"
-    STACK_FILENAME = "exception_frames.filename"
-    STACK_PACKAGE = "exception_frames.package"
-    STACK_MODULE = "exception_frames.module"
-    STACK_FUNCTION = "exception_frames.function"
-    STACK_IN_APP = "exception_frames.in_app"
-    STACK_COLNO = "exception_frames.colno"
-    STACK_LINENO = "exception_frames.lineno"
-    STACK_STACK_LEVEL = "exception_frames.stack_level"
-    CONTEXTS_KEY = "contexts.key"
-    CONTEXTS_VALUE = "contexts.value"
+    """
+    Value is a tuple of (internal Events name, internal Transaction name, external alias)
+    None means the column is not available in that dataset.
+    """
+
+    EVENT_ID = ("event_id", "event_id", "id")
```

Comment:
Why not making these NamedTuples ? It would make this marginally more verbose but the code that reads this enum will be a lot more readable and hard to mess up since you will be referencing the column name by naming the dataset.

Reply:
Thanks, fixed